src/app/components/LanguageAvatar.svelte

```diff
@@ -0,0 +1,69 @@
+<script lang="ts">
+  export let name: string;
+  export let selectedLanguage: string = "";
+
+  let selected: boolean = name === selectedLanguage;
+  let className: string = "avatar";
+
+  /**
+   * Returns the first letter / character of the language.
+   * TODO: Replace with a method that handles abstract cases
+   *
+   * @return {String} - the first character of the language name
+   */
+  const getLanguageShorthand = (): String => {
+    return name.substring(0, 1);
+  };
```

Comment:
I really like this refactor! It has a good name: `getLanguageShorthand()` that clearly indicates its intention, and has a TODO note. I would change the TODO note, though:

```suggestion
  /**
   * Returns the first letter / character of the language.
   * TODO: Handle non-BMP cases like "𐐆" for "𐐆𐑍𐑀𐑊𐐮𐑇" ("English" in the Deseret alphabet)
   * TODO: Handle multiple character grapheme clusters like "k̓"
   *
   * @return {String} - the first character of the language name
   */
  const getLanguageShorthand = (): String => {
    return name.substring(0, 1);
  };
```